types/slate/index.d.ts

```diff
@@ -39,6 +39,7 @@ export interface Rules {
     first?: ObjectAndType | ObjectAndType[];
     isVoid?: boolean;
     last?: ObjectAndType | ObjectAndType[];
+    marks?: Immutable.Set<Mark>;
```

Comment:
this should be an Array<{type: string}> instead of `Immutable.Set<Mark>;`

Reply:
actually it should be `Array<{ type: ((type: string) => boolean) | string }>` as it now receives a function as well

Reply:
right @isubasti , I think the function support is from this PR https://github.com/ianstormtaylor/slate/pull/2151